src/System.Linq/src/System/Linq/Select.SpeedOpt.cs

```diff
@@ -171,7 +171,7 @@ private sealed partial class SelectRangeIterator<TResult> : Iterator<TResult>, I
             public SelectRangeIterator(int start, int end, Func<int, TResult> selector)
             {
                 Debug.Assert(start < end);
-                Debug.Assert((end - start) <= int.MaxValue);
+                Debug.Assert(unchecked((uint)(end - start) <= (uint)int.MaxValue));
```

Comment:
```suggestion
                Debug.Assert((uint)(end - start) <= (uint)int.MaxValue);
```

Reply:
OK, I will force push to change the commit.